internal/pkg/metrics/purge.go

```diff
@@ -57,7 +57,15 @@ func (p *PurgeMetrics) UpdatePurgeTime(duration time.Duration) {
 	p.purgeTimeGauge.Set(duration.Seconds())
 }
 
-func (p *PurgeMetrics) UpdatePurgeError(ctx context.Context, kyma *v1beta2.Kyma, purgeError PurgeError) {
+func (p *PurgeMetrics) SetPurgeError(ctx context.Context, kyma *v1beta2.Kyma, purgeError PurgeError) {
+	p.updatePurgeError(ctx, kyma, purgeError, 1)
+}
+
+func (p *PurgeMetrics) UnsetPurgeError(ctx context.Context, kyma *v1beta2.Kyma, purgeError PurgeError) {
```

Comment:
We should use another approach to unset the metrics, we can remove this metrics entry completely, a metric with 0 doesn't provide too much value, but it still increases the size of the metric, remember we have metrics size limit for Prometheus server to fetch with, so we need to prevent introduce useless metrics.

you can reference [here](https://github.com/kyma-project/lifecycle-manager/blob/f6a6d354f7993da4b730ffac0d14ef6f1f2c6a6e/internal/pkg/metrics/kyma.go#L95) how we clean up other metrics.

Reply:
Done with https://github.com/kyma-project/lifecycle-manager/pull/1392/commits/cf58689cafa1242f0643748ea67fc5caeeff4b02